packages/commons/src/auth/token/interopTokenGenerator.ts

```diff
@@ -1,47 +1,52 @@
 /* eslint-disable max-params */
 import { Algorithm, JwtHeader, JwtPayload } from "jsonwebtoken";
+import {
+  tokenGenerationError,
+  missingRSAKeyError,
+} from "pagopa-interop-models";
 import { v4 as uuidv4 } from "uuid";
+import { logger, signerConfig } from "../../index.js";
 import { userRoles } from "../authData.js";
-import { buildPrivateKeysKidHolder } from "../keys/keyHolder.js";
 import { buildSignerService } from "../signerService.js";
-import { logger } from "../../index.js";
-import { jwtSeedConfig } from "../../config/jwtConfig.js";
-import { InternalToken, TokenSeed } from "./token.js";
+import { InternalToken, TokenSeed, TokenPayloadSeed } from "./token.js";
 
 export type InteropTokenGenerator = {
-  generateInternalToken: () => Promise<InternalToken>;
+  generateInternalToken: (seed: TokenPayloadSeed) => Promise<InternalToken>;
 };
 
-const createInternalTokenWithKid = (
+const createInternalToken = (
   algorithm: Algorithm,
   kid: string,
   subject: string,
   audience: string[],
   tokenIssuer: string,
   validityDurationSeconds: number
 ): TokenSeed => {
-  const issuedAt = new Date();
+  const issuedAt = new Date().getTime() / 1000;
+  const expireAt = validityDurationSeconds * 1000 + issuedAt;
```

Comment:
both `validityDurationSeconds` and `issuedAt` are in seconds, shouldn't this be
```suggestion
  const expireAt = validityDurationSeconds + issuedAt;
```
?

Reply:
Good catch, it's more convenient to uniform all time using second.